### PR TITLE
Integrate Jacoco plugin to calculate unit tests coverage

### PR DESCRIPTION
**Pull request by tharindu-bandara**
## Purpose
Generate unit test coverage information

## Goals
Invoking Jacoco Maven plugin to generate coverage report for 'components/org.wso2.carbon.identity.scim.common'

## Approach
* Adding Jacoco Maven plugin in the parent pom and inheriting it in the  'components/org.wso2.carbon.identity.scim.common' module's pom file
* Adding Maven Surefire plugin and configure it to pick up Jacoco plugin's agent

## User stories
N/A

## Release note
N/A

## Documentation
N/A

## Training
N/A

## Certification
N/A

## Marketing
N/A

## Automation tests
N/A

## Security checks
N/A

## Samples
N/A

## Related PRs
N/A

## Migrations (if applicable)
N/A

## Test environment
N/A
 
## Learning
N/A

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -328,9 +304,65 @@
                         </execution>
                     </executions>
                 </plugin>
+                <plugin>
+                    <groupId>org.jacoco</groupId>
+                    <artifactId>jacoco-maven-plugin</artifactId>
+                    <version>0.8.0</version>
```
Comment (pulasthi7) [inline pom.xml]: Move these versions to properties, similar to the other plugins

File: pom.xml [hunk 2]
```diff
@@ -328,9 +304,65 @@
...
+                                <rules>
+                                    <!-- implementation is needed only for Maven 2 -->
+                                    <rule implementation="org.jacoco.maven.RuleConfiguration">
+                                        <element>BUNDLE</element>
+                                        <limits>
+                                            <!-- implementation is needed only for Maven 2 -->
+                                            <limit implementation="org.jacoco.report.check.Limit">
+                                                <counter>LINE</counter>
+                                                <value>COVEREDRATIO</value>
+                                                <minimum>0.0</minimum>
```
Comment (pulasthi7) [inline pom.xml]: Shall we have this at component level? Each component will have different coverage limit, at least in the initial phrase till we get all to a global limit

File: pom.xml (hunk 1 above)
Comment (tharindu-bandara) [reply pom.xml]: Done.

File: pom.xml (hunk 2 above)
Comment (tharindu-bandara) [reply pom.xml]: Sure. Coverage check rule moved to the child pom.

File: components/org.wso2.carbon.identity.scim.common/pom.xml [hunk 3]
```diff
@@ -154,7 +147,39 @@
                     </instructions>
                 </configuration>
             </plugin>
+            <plugin>
+                <groupId>org.jacoco</groupId>
+                <artifactId>jacoco-maven-plugin</artifactId>
+	        <executions>
```
[the remark this answers is not in the document]
Comment (tharindu-bandara) [reply components/org.wso2.carbon.identity.scim.common/pom.xml]: Made the necessary modification.

File: components/org.wso2.carbon.identity.scim.common/pom.xml [hunk 4]
```diff
@@ -1,5 +1,4 @@
-<?xml version="1.0" encoding="UTF-8"?>
-<!--
+<?xml version="1.0" encoding="UTF-8"?><!--
```
Comment (madurangasiriwardena) [inline components/org.wso2.carbon.identity.scim.common/pom.xml]: Shouldn't the comment starting tag be starting from a new line? Check and fix in all the places

Comment (tharindu-bandara) [reply components/org.wso2.carbon.identity.scim.common/pom.xml]: Fixed in db4763aec62900ad194b54c662e56c31b66f2575.

**Pull request merged**